client/boot/index.js

```diff
@@ -52,6 +53,9 @@ function init() {
 
 	debug( 'Starting Calypso. Let\'s do this.' );
 
+	// prune sync-handler records
+	syncHandler.pruneRecordsFrom();
```

Comment:
I'm not sure about this. is it enough clarify that in the README doc ? Or maybe add a comment. I wouldn't like pass an argument here.